src/Quaternion.jl

```diff
@@ -381,3 +383,23 @@ function slerp(qa::Quaternion{T}, qb::Quaternion{T}, t::T) where {T}
         qa.v3 * ratio_a + qm.v3 * ratio_b,
     )
 end
+
```

Comment:
I found the following `MethodError`.
```julia
julia> sylvester(Quaternion(1,2,3,4),0,0)
ERROR: MethodError: no method matching sylvester(::Quaternion{Int64}, ::Int64, ::Int64)
Closest candidates are:
  sylvester(::Union{Real, Complex}, ::Union{Real, Complex}, ::Union{Real, Complex}) at ~/julia/julia-1.7.1/share/julia/stdlib/v1.7/LinearAlgebra/src/dense.jl:1608
  sylvester(::Quaternion, ::Quaternion, ::Quaternion) at ~/.julia/dev/Quaternions/src/Quaternion.jl:387
Stacktrace:
 [1] top-level scope
   @ REPL[3]:1
```

We can resolve this error by these promotion:
```suggestion

sylvester(a::Quaternion, b::Union{Real,Quaternion}, c::Union{Real,Quaternion}) = sylvester(promote(a,b,c)...)
sylvester(a::Union{Real,Quaternion}, b::Quaternion, c::Union{Real,Quaternion}) = sylvester(promote(a,b,c)...)
sylvester(a::Union{Real,Quaternion}, b::Union{Real,Quaternion}, c::Quaternion) = sylvester(promote(a,b,c)...)
```
Note that these promotions seems able to replaced with
```julia
sylvester(a::Union{Real,Quaternion}, b::Union{Real,Quaternion}, c::Union{Real,Quaternion}) = sylvester(promote(a,b,c)...)
```
but this may overwrite [the original method `sylvester(a::Real, b::Real, c::Real)`](https://github.com/JuliaLang/julia/blob/df2abc6d77a72bde86c41373488a515f55251374/stdlib/LinearAlgebra/src/dense.jl#L1617).

Reply:
I'm not sure we need to deal with `Complex` number with the promotion. (cf #62)

Reply:
The specified promotions left some ambiguities, and by encoding all the methods directly, we were able to simplify quite a bit. e.g. anytime `a` and `b` commute with `x` (i.e. are real), we can compute with many fewer operations.